### PR TITLE
Gets Sunnyvale Ready to Run [Ready to TM]

### PR DESCRIPTION
**Pull request by TripleShades**
## About The Pull Request
Does what it says in the title.
This PR is just to get the map in a playable state, further work will be needed to bring it up to what is envisioned for it.

## Why It's Good For The Game
Since Yuma is going to be out of rotation, this was the viable alternative.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
add: Added a mini elevator "shuttle" to be used by Enclave
add: Added a small oasis PoI
add: Prospector's workshop to town
add: Missing spawn points from various factions
add: Second floor to Mayor's Office
add: Fight ring outside of Town Workshop
add: Gallows outside of Chapel
add: Mining areas to NCR, Legion, Vault and Town
add: Missing faction tcomms machines
add: Ladder to Enclave base in place of the elevator (temporary measure)
tweak: Various scenery additions and tweaks
tweak: Bank is now Mayor's Office
tweak: All NCR stairs now work
tweak: Town farm layout
tweak: Enclave roles shrunk down to make it more in-line with vision
tweak: Vault is now a joinable faction
tweak: Surface-2 tcomms now uses a single machine to connect to actual tcomms in Underground-1
del: Removed coin press from Bank (it is broken currently)
del: Unused faction tcomms machines
del: Enclave base elevator console and dock point (could not get it to work)
fix: NCR airlock access in two places
fix: Above-3 areas not being correct in parts of the map that had rooftops
fix: Above-3 not having tcomms
code: Added Enclave mini elevator
code: Followers radio frequency
/:cl:

### CODE REVIEW

**Comment by hermaplusplus** [comment]
This PR needs to target the `maps/sunnyvale` branch.

**Review by TripleShades — COMMENTED** [review]
god I hate elevators
wait why am I allowed to review my own pr

File: code/controllers/subsystem/maptype.dm [hunk 1]
```diff
@@ -17,5 +17,5 @@ SUBSYSTEM_DEF(maptype)
 	//The faction tag is unused currently, but the map might break without it
 
 	//All the map tags that delete all jobs and replace them with others.
-	var/list/clearmaps = list()
+	var/list/clearmaps = list("sunnyvale")
```
Comment (Kitsunemitsu) [inline code/controllers/subsystem/maptype.dm:20]: Please don't touch this.
The clear maps variable is to clear ALL jobs and replace with a whole new set.

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -695,7 +695,7 @@
 		dat += "</tr></table>"
 
 	//BoS (Steel Blue)
-		dat += "<table cellpadding='1' cellspacing='0' width='100%'>"
+	/*	dat += "<table cellpadding='1' cellspacing='0' width='100%'>"
```
Comment (Kitsunemitsu) [inline code/modules/admin/topic.dm:698]: No need to comment this out.
Better to have Legacy code that isn't used than need it and have to call the right guy to re-enable it

File: code/modules/jobs/job_types/enclave.dm [hunk 3]
```diff
@@ -91,7 +91,7 @@
 /datum/job/enclave/enclavelt
 	title = "Enclave Lieutenant"
 	flag = F13USLT
-	total_positions = 1
+	total_positions = 0
```
Comment (Kitsunemitsu) [inline code/modules/jobs/job_types/enclave.dm:94]: Don't do this.
God PLEASE don't do this.

Make the enclave template the "enclave-military" maptype, and put into an enclave folder.
You can then make your new enclave file.
This is for legacy and cleanliness sake.

If you need help doing this, Just ask me and I can do it tmrw.

File: code/modules/jobs/job_types/vault.dm [hunk 4]
```diff
@@ -12,8 +12,6 @@ here's a tip, go search DEFINES/access.dm
 	department_flag = VAULT
 	exp_type = EXP_TYPE_VAULT
 	faction = FACTION_VAULT
-	maptype = "vault"
```
Comment (Kitsunemitsu) [inline code/modules/jobs/job_types/vault.dm:15]: Set maptype to Sunnyvale. Already done in a fix PR. Irrelevant.

File: code/modules/jobs/jobs.dm [hunk 5]
```diff
@@ -304,9 +304,9 @@ GLOBAL_LIST_INIT(exp_jobsmap, list(
 	EXP_TYPE_SILICON = list("titles" = list("AI","Cyborg")),
 	EXP_TYPE_SERVICE = list("titles" = civilian_positions),
 
-	EXP_TYPE_FALLOUT = list("titles" = brotherhood_positions | oasis_positions | legion_positions | ncr_positions | vault_positions | wasteland_positions | tribal_positions | followers_positions | enclave_positions),
```
Comment (Kitsunemitsu) [inline code/modules/jobs/jobs.dm:307]: Don't touch this.
This counts up EXP for the sake of a general timelock. If you remove BOS from here it's *just* a pain in the ass.
If you keep it in, Nothing bad happens.

**Review by Kitsunemitsu — CHANGES_REQUESTED** [review]
Serious changes requested.

File: code/controllers/subsystem/maptype.dm (hunk 1 above)
Comment (TripleShades) [reply code/controllers/subsystem/maptype.dm:20]: I only copied what I saw in the Reno pr, so shrug.
I'll revert it back to what it was.

File: code/modules/jobs/job_types/enclave.dm (hunk 3 above)
Comment (TripleShades) [reply code/modules/jobs/job_types/enclave.dm:94]: I don't understand what you're asking, so yes please help me tomorrow.

File: code/modules/jobs/job_types/vault.dm (hunk 4 above)
Comment (TripleShades) [reply code/modules/jobs/job_types/vault.dm:15]: Forgot to set this to Sunnyvale when debugging why Vault roles weren't showing up.

File: code/modules/jobs/jobs.dm (hunk 5 above)
Comment (TripleShades) [reply code/modules/jobs/jobs.dm:307]: That makes sense, I was just removing everything I thought would mess with the job select screen, but I should have read the variable name.

File: code/modules/jobs/job_types/enclave.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/modules/jobs/job_types/enclave.dm:94]: Also this doesn't even remove the slots, just the readyup slots

File: code/modules/jobs/jobs.dm (hunk 5 above)
Comment (Kitsunemitsu) [reply code/modules/jobs/jobs.dm:307]: Job select screen will be set in a fix PR coming up later tnight.
That's actually part of where I specialize.

File: code/modules/jobs/job_types/enclave/enclave.dm [hunk 6]
```diff
@@ -328,7 +328,7 @@
 	total_positions = 2
 	spawn_positions = 2
 	description = "You are an operative for the remnants of the Enclave. You, unlike the normal privates, have recieved specialist training in either engineering or medicine."
-	supervisors = "The Lieutenant and the Sergeants."
```
Comment (Kitsunemitsu) [inline code/modules/jobs/job_types/enclave/enclave.dm:331]: Revert this change

File: code/modules/jobs/job_types/enclave/enclave-sunnyvale.dm [hunk 7]
```diff
@@ -0,0 +1,577 @@
+//It looks like var/faction controls what becomes visible on setup. Should likely be fixed or something, but I'm not doing it.
+/datum/job/enclave
+	department_flag = ENCLAVE
+	selection_color = "#323232"
+	faction = FACTION_ENCLAVE
+	exp_type = EXP_TYPE_FALLOUT
```
Comment (Kitsunemitsu) [inline code/modules/jobs/job_types/enclave/enclave-sunnyvale.dm:6]: Add 
`maptype = "sunnyvale"`

**Review by Kitsunemitsu — CHANGES_REQUESTED** [review]
Just some tiny fixes you need to do.

**Comment by Kitsunemitsu** [comment]
Also don't forget to strip down the new enclave file

**Comment by TripleShades** [comment]
> Also don't forget to strip down the new enclave file

Oh yeah that'd probably be a good idea huh, so it's less confusing/bloated.

**Review by Kitsunemitsu — APPROVED** [review]
Approved. Waiting on the head maintainer.
